server/src/main/java/com/cloud/consoleproxy/AgentHookBase.java

```diff
@@ -213,10 +212,14 @@ public void startAgentHttpHandlerInVM(StartupProxyCommand startupCmd) {
 
             byte[] ksBits = null;
 
-            String consoleProxyUrlDomain = _configDao.getValue(Config.ConsoleProxyUrlDomain.key());
-            String consoleProxySslEnabled = _configDao.getValue(ConsoleProxyManager.ConsoleProxySslEnabled.key());
-            if (!StringUtils.isEmpty(consoleProxyUrlDomain) && !StringUtils.isEmpty(consoleProxySslEnabled)
-                    && consoleProxySslEnabled.equalsIgnoreCase("true")) {
+            HostVO consoleProxyHost = findConsoleProxyHost(startupCmd);
+
+            assert (consoleProxyHost != null);
```

Comment:
when are we running with asserts? maybe we can log a fatal error if `consoleProxyHost` is null?

Reply:
I just used the historic code - but I'll change the approach.. to add a meaningful error / warning ..